csharp/lang/correctness/regioninfo/regioninfo-interop.yaml

```diff
@@ -0,0 +1,36 @@
+rules:
+- id: correctness-regioninfo-interop
+  patterns:
+  - pattern-either:
+    - pattern: |
+        $WRITER.Write($RI);
+    - pattern: |
+        $WRITER.WriteAsync($RI);
+    - pattern: |
+        $WRITER.WriteLine($RI);
+    - pattern: |
+        $WRITER.WriteLineAsync($RI);
+  - pattern-inside: |
+      RegionInfo $RI = new RegionInfo($REGION);
+      ...
+      using($PIPESTREAM $P = ...){
+        ...
+      }
+  - metavariable-regex:
+      metavariable: $REGION
+      regex: ^"\w{2}"$
+  - metavariable-regex:
+      metavariable: $PIPESTREAM
+      regex: (Anonymous|Named)Pipe(Server|Client)Stream
+  message: Potential inter-process write of RegionInfo $RI via $PIPESTREAM $P that
```

Comment:
What does one do to fix this?

Reply:
you properly initialize the `RegionInfo` with correct metadata - the two-character code does an incomplete initialization of the object that loses data when transmitted between processes.

Reply:
^ That's a great explanation :D

Most rules are OK so long as the receiver of the findings understands the issue and can make an informed choice about it. Thanks!